### PR TITLE
fix(sankey): fix other state color is 'target/source/gradient'

### LINKED ISSUE #18804: [Bug] Sankey `emphasis.lineStyle.color: 'target'` not work
### Version

5.4.2

### Link to Minimal Reproduction

https://echarts.apache.org/examples/zh/editor.html?c=sankey-simple&code=PYBwLglsB2AEC8sDeAoWsDOBTAThLGAXMmurGAJ4hbEDkGAhtANZYW0A0p6AJg2A2IBtbmVRkJsaAwC2NWLQa1R6AL5dJJTemly6AI04qyEMFhkBlSgBt547egDGwa8Bx0cWHsoerjfiQBdDTJrCBYiWBFNe00MYABXHEd5RSMHARwAcywwA3TtADcGawT5AFYQzTDoLCsKW2JY7WdXdwV4pJSfbXVjdHMQAAsGDAhI5urwups7fslWtzpMnLAe33mAyT6g0j9VAG4gA

### Steps to Reproduce

1. Hover the line between a and b

### Current Behavior

The color of the line between a and b is gray.

### Expected Behavior

The color of the line between a and b is red(the color of b).

### Environment

```markdown
- OS: MacOS 13.3
- Browser: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36
- Framework: -
```

### Any additional comments?

_No response_

### PR DESCRIPTION
**Pull request by linghaoSu**
## Brief Information

This pull request is in the type of:

- [x] bug fixing
- [ ] new feature
- [ ] others

### What does this PR do?

fix sankey emphsis lineStyle color is not a normal color, using node color(`source`, `target`, `gradient`)

### Fixed issues

- fix #18804

### Before: What was the problem?

not parse color in emphsis lineStyle (source,target or gradient), so using a default color

image

### After: How does it behave after the fixing?

Using logic in normal state to parse color(`target`, `source` or `gradient`), apply that to other state.

image
image
image

## Document Info

One of the following should be checked.

- [x] This PR doesn't relate to document changes
- [ ] The document should be updated later
- [ ] The document changes have been made in apache/echarts-doc#xxx

### ZRender Changes

- [ ] This PR depends on ZRender changes (ecomfe/zrender#xxx).

### Related test cases or examples to use the new APIs

N.A.

### Merging options

- [ ] Please squash the commits into a single one when merging.

### CODE REVIEW

File: src/chart/sankey/SankeyView.ts [hunk 1]
```diff
@@ -195,30 +196,32 @@ class SankeyView extends ChartView {
...
+            //                     color: sourceColor,
+            //                     offset: 0
+            //                 }, {
+            //                     color: targetColor,
+            //                     offset: 1
+            //                 }]
+            //             );
+            //         }
+            // }
+
```
Comment (plainheart) [inline src/chart/sankey/SankeyView.ts:224]: Just please clean the commented code fragement.

File: src/chart/sankey/SankeyView.ts [hunk 2]
```diff
@@ -30,6 +30,7 @@ import { RectLike } from 'zrender/src/core/BoundingRect';
 import { setLabelStyle, getLabelStatesModels } from '../../label/labelStyle';
 import { getECData } from '../../util/innerStore';
 import { isString, retrieve3 } from 'zrender/src/core/util';
+import { GraphEdge } from '../../data/Graph';
```
Comment (plainheart) [inline src/chart/sankey/SankeyView.ts:33]: It's better if we import the `GraphEdge` type explicitly with the `type` keyword.

File: src/chart/sankey/SankeyView.ts (hunk 1 above)
Comment (linghaoSu) [reply src/chart/sankey/SankeyView.ts]: removed commented code

File: src/chart/sankey/SankeyView.ts (hunk 2 above)
Comment (linghaoSu) [reply src/chart/sankey/SankeyView.ts]: import with type

**Review by plainheart — APPROVED** [review]

**Pull request merged**